### ISSUE TITLE
reportUnusedImport incorrectly reported

### ISSUE THREAD

**Issue by mehdigmira** [microsoft/pyright#6645]
**Describe the bug**
reportUnusedImport is reported in the code below, where it should not be.

**Code or Screenshots**

```python
from __future__ import annotations

import math
from typing import Any

import pytz  # reportUnusedImport reported here


def test( x: Any, y: Any, z: Any, b:Any):
    a = x
    while a < y:
        if a not in z:  
            b = int(math.ceil(b / 30.0))
        else:  
            b = z[a]

        if b:
            return a.astimezone(pytz.utc).time().isoformat(timespec="seconds")

```

**VS Code extension or command-line**
pyright: 1.338

**Comment by erictraut** [comment]
Thanks for the bug report. I'm able to repro the problem, and I'll investigate further.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.339, which I just published. It will also be included in a future release of pylance.
